xarray/core/dataarray.py

```diff
@@ -840,6 +840,99 @@ def swap_dims(self, dims_dict):
         ds = self._to_temp_dataset().swap_dims(dims_dict)
         return self._from_temp_dataset(ds)
 
+    def expand_dims(self, dim=None, axis=None):
```

Comment:
> What is the reason to implement in xarray.Dataset? Does it mean that we would also support Dataset.expand_dims?

Yes, exactly.

> It would be reasonable to raise an ValueError if any of data_var has smaller dimension that is required by axis?

Yes, of course. In practice, I suspect most people will either keep the default (to expand dims at the start) or use negative integers (to expand dims at the end).